qiime2/metadata/metadata.py

```diff
@@ -877,7 +876,11 @@ def __init__(self, series):
                 "%s %r does not support a pandas.Series object with dtype %s" %
                 (self.__class__.__name__, series.name, series.dtype))
 
+        series.index = series.index.str.strip()
+        series.name = series.name.strip()
         self._series = self._normalize_(series)
+        if isinstance(self, CategoricalMetadataColumn):
```

Comment:
This particular check would be unnecessary if the below branch was a part of `_normalize_` as method dispatch solves the "who" we are problem.

Reply:
But we want to normalize the `series` prior to stripping don't we? I've been poking at it, and unless I'm missing something this isn't really doable in `_normalize_` given its current implementation with the inner function and all

Reply:
Wait I've had an epiphany and realized I am dumb, nevermind.